src/python/pants/bin/daemon_pants_runner.py

```diff
@@ -257,21 +253,18 @@ def run(self):
         # Clean global state.
         clean_global_runtime_state(reset_subsystem=True)
 
-        # Setup the Exiter's finalizer.
-        self._exiter.set_finalizer(finalizer)
-
         # Otherwise, conduct a normal run.
-        runner = LocalPantsRunner.create(
-          PantsRunFailCheckerExiter(),
-          self._args,
-          self._env,
-          target_roots,
-          graph_helper,
-          options_bootstrapper,
-        )
-        runner.set_start_time(self._maybe_get_client_start_time_from_env(self._env))
-
-        runner.run()
+        with ExceptionSink.exiter_as(PantsRunFailCheckerExiter):
```

Comment:
Here, the constructor is the `new_exiter_fun`. Although, now that you mention it, this is totally incorrect, because the constructor doesn't take a previous exiter, so this should be `(lambda _: PantsRunFailCheckerExiter()).

Good catch, thanks!